astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,271 @@
+                       'earth', 'moon', 'mars', 'jupiter', 'saturn', 'uranus', 
+                       'neptune', 'pluto']
+BODIES = {'sun': [(0, 10)],
+          'mercury': [(0, 1), (1, 199)],
+          'venus': [(0, 2), (2, 299)],
+          'earth-moon-barycenter': [(0, 3)],
+          'earth':  [(0, 3), (3, 399)],
+          'moon': [(0, 3), (3, 301)],
+          'mars': [(0, 4)],
+          'jupiter': [(0, 5)],
+          'saturn': [(0, 6)],
+          'uranus': [(0, 7)],
+          'neptune': [(0, 8)],
+          'pluto': [(0, 9)]}
+
+
+def _download_spk_file(show_progress=True):
+    """
+    Get the Satellite Planet Kernel (SPK) file `de430.bsp` from NASA JPL.
+
+    Download the file from the JPL webpage once and subsequently access a
+    cached copy. This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+
+    .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
+    """
+    de430_url = ('http://naif.jpl.nasa.gov/pub/naif/'
+                 'generic_kernels/spk/planets/de430.bsp')
+    return download_file(de430_url, cache=True, show_progress=show_progress)
+
+
+def _get_kernel(*args, **kwargs):
+    """
+    Try importing jplephem, download/retrieve from cache the Satellite Planet
+    Kernel.
+    """
+    global KERNEL
+
+    try:
+        from jplephem.spk import SPK
+    except ImportError:
+        raise ImportError("Solar system ephemeris calculations depend on"
+                          "the jplephem package")
+
+    if KERNEL is None:
+        KERNEL = SPK.open(_download_spk_file())
+    return KERNEL
+
+
+def get_barycentric_body_position(time, body):
+    """
+    Calculate the barycentric position of solar system body ``body`` in cartesian coordinates.
+
+    Uses ``jplephem`` with the DE430 kernel.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body : str
+        The solar system body to calculate. 
+        
+        The allowed values for ``body`` can be found in 
+        `~astropy.coordinates.SOLAR_SYSTEM_BODIES`.
+
+    Returns
+    -------
+    cartesian_position : `~astropy.coordinates.CartesianRepresentation`
+        Barycentric (ICRS) position of the body in cartesian coordinates
+
+    Notes
+    -----
+
+    """
+    #lookup chain
+    chain = BODIES[body.lower()]
+
+    kernel = _get_kernel()
+
+    # are all the kernel pairs in this kernel?
+    valid_chain = np.all([c in kernel.pairs.keys() for c in chain])
+    if not valid_chain:
+        raise ValueError("Postion of this body cannot be calculated using jpl kernel")
+
+    jd1, jd2 = get_jd12(time, 'tdb')
+
+    cartesian_position_body = sum([kernel[pair].compute(jd1, jd2) for pair in chain])
+
+    barycen_to_body_vector = u.Quantity(cartesian_position_body, unit=u.km)
+    return CartesianRepresentation(barycen_to_body_vector)
+
+
+def _get_earth_body_vector(time, body, earth_time=None):
```

Comment:
I'd probably move this into `apparent_position` below (but we can do it later if need be).